fixes#2358-skin-tissues

paulacalipho opened the pull request:
fixes #2358 -skin-tissues

shawntanzk commented:
@paulacalipho - just wondering what the use cases for these terms are and if its possible they could be represented as "skin and ("part of" some _____) instead? I do understand that some use cases require an individual term, but just checking.

paulacalipho commented:
@shawntanzk these terms are requested to map the tissues used to create cell lines. Below I show you the cell lines derived for each case. 
@shawntanzk I do not understand your 2nd comment because I tried to create these terms as "skin and ("part of" some _____), but I probably did something wrong

**skin of breast**: https://web.expasy.org/cellosaurus/CVCL_2339; https://web.expasy.org/cellosaurus/CVCL_2346; https://web.expasy.org/cellosaurus/CVCL_2330; https://web.expasy.org/cellosaurus/CVCL_X774
skin of buttock	https://web.expasy.org/cellosaurus/CVCL_0J95; https://web.expasy.org/cellosaurus/CVCL_F485; https://web.expasy.org/cellosaurus/CVCL_L473; https://web.expasy.org/cellosaurus/CVCL_7289
**skin of lateral lumbar region of abdomen**	https://web.expasy.org/cellosaurus/CVCL_LH33
**skin of sacral region**: https://web.expasy.org/cellosaurus/CVCL_2Q25
**skin of clavicle region**: https://web.expasy.org/cellosaurus/CVCL_IR20
**skin of iliac crest region**: https://web.expasy.org/cellosaurus/CVCL_GQ27; https://web.expasy.org/cellosaurus/CVCL_GQ51
**skin of temple**: https://web.expasy.org/cellosaurus/CVCL_G300; https://web.expasy.org/cellosaurus/CVCL_A5VI
**skin of external genitalia**: https://web.expasy.org/cellosaurus/CVCL_IL08; https://web.expasy.org/cellosaurus/CVCL_IL12

shawntanzk commented:
> I do not understand your 2nd comment because I tried to create these terms as "skin and ("part of" some _____), but I probably did something wrong

oh no, it's not you doing something wrong, its asking if instead of creating a term, can it be annotated with a more complex thing like "skin and ("part of" some _____)" (using two uberon terms) instead of one. My worry with terms like these is that it ends up making the ontology "incomplete" - there would be skin of many terms and getting them all in is difficult. But I do know our ontologies have many terms like these cause users need them, so it was just me checking.

paulacalipho commented:
@shawntanzk OK, now I understand what you mean. For some cases I guess one could combine terms. But for others, they may deserve being different terms. For example: the skin of the foot sole has probably different characteristics than the skin of nose, and may deserve being different entities.

paulacalipho commented:
@shawntanzk the terms I created I think are OK. Then there are all the "gustatory cortex" annotations that are as "removed", but I do not know why.

shawntanzk commented:
> Then there are all the "gustatory cortex" annotations that are as "removed", but I do not know why.

Oh I just merged that in, I'm guessing its just about updating the branch with master and it will be alright :)

paulacalipho commented:
@shawntanzk Sorry, I was not looking at the simple diff. Will check and let you know

paulacalipho commented:
@shawntanzk The classes that have been changed as the result of my additions are NOT correct. For example, it added "ala of nose" as subclass of "temple" [UBERON_8480027]. I do not understand why. What should I do ? Restart everything from scratch?

shawntanzk commented:
> @shawntanzk The classes that have been changed as the result of my additions are NOT correct.

This is a reasoned diff - its basically what the ontology would look like with an ELK reasoner on. Restarting wouldn't fix it - what you need to do instead is figure out why it is being reasoned that way (eg why ala of nose is reasoned to be a subclass of temple). You can do this by opening up the ontology in protege, reasoning it, then going to the the terms that has inferred subclass and clicking on the ? at that inferred axiom. This will give you a reason why. From there we need to fix axioms that causes the wrongly inferred class. Do let me know how you go with that - happy to help.

paulacalipho commented:
@shawntanzk I probably wrongly defined temple as 'lateral structure' and ('part of' some head), which now I realize is wrong. Should be "anatomical structure" and ('part of' some head) correct?

shawntanzk commented:
> @shawntanzk I probably wrongly defined temple as 'lateral structure' and ('part of' some head), which now I realize is wrong. Should be "anatomical structure" and ('part of' some head) correct?

Give it a try and run the reasoner and see if it fixes it :) one you push the changes to this branch, the GitHub actions will run again and give us an updated reasoned diff which we can check again.

paulacalipho commented:
@shawntanzk Protege is hanging. I will continue on Thursday, and go back to you. Thanks for your help

paulacalipho commented:
@shawntanzk I changed the Equivalence to "anatomical structure" and ('part of' some head)" and run the Reasoner again and since 1h30min there is a window in Protege that stays in "Incremental Overdeletion Pruning". Is this normal?

shawntanzk commented:
> @shawntanzk I changed the Equivalence to "anatomical structure" and ('part of' some head)" and run the Reasoner again and since 1h30min there is a window in Protege that stays in "Incremental Overdeletion Pruning". Is this normal?

defs not, but also that equivalence looks wrong - that basically says that everything that is part of the head is a subclass of your term

shawntanzk commented on src/ontology/uberon-edit.obo:
```diff
@@ -224667,6 +224685,71 @@ is_a: UBERON:0002553 ! anatomical cavity
+
+[Term]
+id: UBERON:8480027
+name: temple
```
@bvarner-ebi - could you help a bit with this. I saw that there is temporalis muscle in uberon (http://purl.obolibrary.org/obo/UBERON_0001598) and this should have some relation to temple (I assume), but not 100% sure how it actually relates - would you mind adding the appropriate relation please? thanks :)

bvarner-ebi replied:
@shawntanzk, I would be inclined to model the muscle wrt to bones. I would not add a relation between the muscle and a body region unless you see examples of this in Uberon.

shawntanzk replied:
perhaps then adding bone relation to temple? Just thought ideally that muscle should have a relation to temple given the name, but if you think otherwise that's fine too :)

bvarner-ebi replied:
> perhaps then adding bone relation to temple?

This sounds reasonable to me since the temple exists through the presence of the fused bones, not the presence of the muscle.

bvarner-ebi replied:
You could add gloss regarding the muscle if you're keen on mentioning it.

bvarner-ebi commented on src/ontology/uberon-edit.obo:
```diff
@@ -224667,6 +224685,71 @@ is_a: UBERON:0002553 ! anatomical cavity
+[Term]
+id: UBERON:8480027
+name: temple
+def: "The region where the frontal, parietal, temporal, and sphenoid bones fuse. It is located on the side of the head behind the eye and between the forehead and the ear." [https://en.wikipedia.org/wiki/Temple_(anatomy)]
```
Consider "posterior" in place of "behind".

paulacalipho replied:
@bvarner-ebi OK, I will correct to "posterior". 
@bvarner-ebi what do you mean with "EDIT: but not sure if this holds true in all species."?

bvarner-ebi replied:
@paulacalipho, it was just a note that I made changes to my original comment. I edited the comment again so it is clearer.

paulacalipho replied:
@bvarner-ebi Do you think that this definition is better? "The region where the frontal, parietal, temporal, and sphenoid bones fuse."

bvarner-ebi replied:
I think that is sufficient. If the second sentence in your original definition is always true, you can keep it as well.

paulacalipho replied:
@bvarner-ebi I will keep only the first part, to be in the safe side, thanks for the input

paulacalipho replied:
@bvarner-ebi I am trying to re-open the branch in protege to correct the definition but I got this error message: 
![image](https://user-images.githubusercontent.com/4089751/188113512-14ab4786-ea43-404c-a47c-f3cf535b077a.png)

shawntanzk replied:
could you run the following in your command line and paste it here please: 
`head uberon-edit.obo`

paulacalipho replied:
@shawntanzk 
![image](https://user-images.githubusercontent.com/4089751/188119661-688d4ed7-20d9-432f-a534-14468536574c.png)

shawntanzk replied:
oh ok, im not sure how windows work lol - maybe just open up uberon-edit.obo in a text editor and copy and paste the first few lines here

paulacalipho replied:
@shawntanzk these are few lines:
format-version: 1.2
subsetdef: added_for_HCA "Classes tagged with this subset property were added upon request from the Human Cell Atlas (HCA)."
subsetdef: cumbo "CUMBO"
subsetdef: cyclostome_subset "cyclostome structures"
subsetdef: defined_by_cytoarchitecture "brain region defined by cyto-architecture"

and then this is what I see in GitHub desktop
![image](https://user-images.githubusercontent.com/4089751/188120933-5c23826c-fc1c-4edc-8281-5a4831b45751.png)

shawntanzk replied:
Oh! you have clashes. 
For now you can just remove all the added lines and it should work :)

paulacalipho replied:
@shawntanzk it works now, many thanks

shawntanzk commented on src/ontology/uberon-edit.obo:
```diff
@@ -224667,6 +224685,71 @@ is_a: UBERON:0002553 ! anatomical cavity
+[Term]
+id: UBERON:8480027
+name: temple
+def: "The region where the frontal, parietal, temporal, and sphenoid bones fuse." [https://en.wikipedia.org/wiki/Temple_(anatomy)]
```
I would prefer the wikipedia link to be instead WikipediaVersioned:Temple_(anatomy)&oldid=1104547137 
This is important for a resource like Wikipedia as it gets updated and we want to be sure where the source is from :)

paulacalipho replied:
@shawntanzk Not sure I understood. Do you want me to replace the wikipedia link with "WikipediaVersioned:Temple_(anatomy)&oldid=1104547137"
I checked the generated file and seems OK to me

shawntanzk replied:
> Do you want me to replace the wikipedia link with "WikipediaVersioned:Temple_(anatomy)&oldid=1104547137"

yes

shawntanzk commented:
@paulacalipho - do help me make sure all the changes to the simple file are intended (I scanned through them and they look right). Other than that just one note on the Wikipedia dbxref on def. Thanks

paulacalipho commented:
@shawntanzk I checked that the file is correct and changed the Wikipedia dbxref on def as you suggested. Many thanks for your review

shawntanzk reviewed (approved):
I'll approve for now, but I'd prefer it if temple had some form of relation to bone that has relation with temporal muscle - I do get that this is a bit more complicated and that you might need help from @bvarner-ebi. Thanks

paulacalipho commented:
Hi @bvarner-ebi I think that "temple" is a region on each side of the head. I do not think is a bone structure. What do you think? Apart from that, should I ask emquardokus as a reviewer for this set of terms?

bvarner-ebi commented:
> Hi @bvarner-ebi I think that "temple" is a region on each side of the head. I do not think is a bone structure. What do you think? Apart from that, should I ask emquardokus as a reviewer for this set of terms?

Hi, @paulacalipho. Firstly, it is great to see all the new terms you are submitting!

Regarding 'temple'... yes, I agree it is an anatomical region. According to your reference the location is defined by "where four skull bones fuse: the [frontal](https://en.wikipedia.org/wiki/Frontal_bone), [parietal](https://en.wikipedia.org/wiki/Parietal_bone), [temporal](https://en.wikipedia.org/wiki/Temporal_bone), and [sphenoid](https://en.wikipedia.org/wiki/Sphenoid_bone)." 

If you cannot readily find a way to add axioms to this term after searching for similar terms in Uberon and seeing how they are modelled, my recommendation is to leave as-is.
My thought: since this is a "region", one has to consider, what are the boundaries, does it also include the muscle, skin, etc. This is why it is not as straightforward as other terms. If you just include the basic, non-controversial information, I think that would be sufficient for now. 

Here is how FMA breaks it down: https://bioportal.bioontology.org/ontologies/FMA/?p=classes&conceptid=http%3A%2F%2Fpurl.org%2Fsig%2Font%2Ffma%2Ffma46450

If you want, you can add this as a discussion item on the Uberon meeting agenda so we can discuss further: https://docs.google.com/document/d/1XvMbNvr0FEsdqGhg79BYCYEHSqUxRHMcvhbGizEAht8/edit.

On another note, what you have is a human-centric definition. This definition would not work, for example, in some insects: [ICHNEUMONID](http://www.amentinst.org/GIN/morphology.php). If the definition and modelling you will be submitting is only applicable to vertebrates, for example, you can consider appending (sensu Vertebrata) to the label or adding axioms to restrict to a particular taxon.

paulacalipho commented:
@bvarner-ebi many thanks for your feedback. I tried to get inspired with the term "cheek" that's why I made it subset of anatomical structure. About the boundaries, I am not sure how to define them. I also thought about to which taxonomy it applies, and I would say Primates, not sure it applies further. Should I add an axiom to restrict to this taxon?
I added (in suggested mode) this item in the agenda. Not sure I added it in the right format and place
Thanks again for your help

bvarner-ebi commented:
Hi, @paulacalipho,
Great, I see your item on the agenda. Please do feel free to join the meeting (joining info at top of the agenda), but understand if it is too late for your time zone.
A primate restriction seems reasonable.

paulacalipho commented:
@bvarner-ebi @shawntanzk Thanks to both of you for the feedback.  Do you agree if I add the issue about primates or vertebrates is the agenda of the Uberon meeting?

shawntanzk commented:
@paulacalipho our meetings are open for anyone to add items so feel free to do so :)

bvarner-ebi commented:
@paulacalipho, suggestion from the Uberon call is to model 'temple' as a region to align with FMA and the rest of your "region" terms.

See [here](https://www.ebi.ac.uk/ols/ontologies/fma/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FFMA_53640&lang=en&viewMode=All&siblings=false) how temple includes some of the adjacent anatomical structures in the FMA modelling:

<img width="1320" alt="Screenshot 2022-09-27 at 10 41 49" src="https://user-images.githubusercontent.com/95643797/192493543-bfd1daa4-4908-490b-85df-3e0f19eca2e4.png">

paulacalipho commented:
@bvarner-ebi sorry for the delay in answering and for not be able to be at the meeting. Should I 
1) create first in UBERON "Subdivision of temporal part of head" and "Auriculotemporal part of head" to then make "Temple" a subclass of them?
2) add only in taxon some Vertebrata?

bvarner-ebi commented:
Hi, @paulacalipho,
 
> 1. create first in UBERON "Subdivision of temporal part of head" and "Auriculotemporal part of head" to then make "Temple" a subclass of them?

I don't think it is necessary to create these new classes. I would proceed similarly with how you created your other anatomical "region" terms.

> 2. add only in taxon some Vertebrata?

@shawntanzk [suggested](https://github.com/obophenotype/uberon/pull/2637#issuecomment-1252296015) not adding the taxon restrictions. Making the intended meaning clear in the text definition would be sufficient for now.

paulacalipho commented:
@bvarner-ebi @shawntanzk I pushed some changes, but I am not sure I fully understood. Let me know

shawntanzk commented:
"This term may apply to vertebrates." defs does not belong in the definition. 

My question would be does this statement

> The region where the frontal, parietal, temporal, and sphenoid bones fuse.

apply to all temples? vertebrate or not? if so then you don't need "This term may apply to vertebrates."
If not I would want to know which part of the definition doesn't apply to vertebrates. 

If there are specific logical axioms that are vertebrate specific, I would also prefer a taxon GCI on that if possible - but I do not see logical axioms for any of the definition, so I think that's fine. 

Hope that helps

paulacalipho commented:
@shawntanzk the definition applies to all temples. But I do not know if all taxons have "temple"

shawntanzk commented:
I see @bvarner-ebi has commented on this

> On another note, what you have is a human-centric definition. This definition would not work, for example, in some insects: [ICHNEUMONID](http://www.amentinst.org/GIN/morphology.php). If the definition and modelling you will be submitting is only applicable to vertebrates, for example, you can consider appending (sensu Vertebrata) to the label or adding axioms to restrict to a particular taxon.

I do see that in insects this is (probably) different, though I would hesitate to make an exception for that given it seems to have another name in insect and I'm guessing that the use of "temple" is more as a reference to vertebrates. Maybe @gouttegd or any other insect people could comment?

Perhaps instead of adding "This term may apply to vertebrates.", a comment stating that this term does not apply to insects temple? and instead of a 'only in taxon' restriction, a 'never in taxon' for insects - (not sure what exact ncbitaxon term this would be, you might have to find it). 

ah the joys of a multispecies anatomy ontology lol

bvarner-ebi commented:
@paulacalipho, I agree with @shawntanzk's comment:

>Perhaps instead of adding "This term may apply to vertebrates.", a comment stating that this term does not apply to insects temple? 

Specifically:
-  I suggest removing "This term may apply to vertebrates." from the text definition
- add a comment that this term does not refer to the "gena" of insects, which temple in that instance is a synonym (listed here: http://portal.hymao.org/projects/32/public/ontology_class/show_expanded/482).

paulacalipho commented:
@bvarner-ebi @shawntanzk many thanks. May I close the pull request now?

shawntanzk commented on src/ontology/uberon-edit.obo:
```diff
@@ -224667,6 +224685,73 @@ is_a: UBERON:0002553 ! anatomical cavity
+id: UBERON:8480027
+name: temple
+def: "The region where the frontal, parietal, temporal, and sphenoid bones fuse." [WikipediaVersioned:Temple_(anatomy)&oldid=1104547137]
+comment: This term does not refer to the "gena" of insects, which temple in that instance is a synonym (listed here: http://portal.hymao.org/projects/32/public/ontology_class/show_expanded/482).
```
instead of listed here, I think make this a dbxref

paulacalipho replied:
@shawntanzk I do not understand where to put this xref. Because we are saying that temple defined here in UBERON:8480027 does not refer to the temple refered in the xref mentioned in the comment

shawntanzk commented on src/ontology/uberon-edit.obo:
```diff
@@ -224667,6 +224685,73 @@ is_a: UBERON:0002553 ! anatomical cavity
+property_value: dcterms-date 2022-08-29T14:19:16Z xsd:dateTime
+
+[Term]
+id: UBERON:8480027
```
given all the discussion on this,  I think it would be wise to be a seeAlso annotation and link it to this ticket

bvarner-ebi commented:
Hi, @paulacalipho, since this pull request was opened, a new Uberon release came out. There are now merge conflicts since the Uberon master branch has changed.

@shawntanzk, can Paula resolve this by following the instructions here?
https://oboacademy.github.io/obook/howto/fixing-conflicts/

If not, can you advise the best way to proceed?

shawntanzk commented:
> @shawntanzk, can Paula resolve this by following the instructions here?
https://oboacademy.github.io/obook/howto/fixing-conflicts/

looks like only the -edit file is conflicted, nothing much to do with the release actually - just that there was changes committed to the edit file during this period. Those instructions should work, but do let us know if you face any issues :)

paulacalipho commented:
@shawntanzk I am trying to edit in Notepad++ (I do not know what I did but I can not open with Protege) and what is missing is the UBERON ID of one of the entries (tendon of semitendinosus). How to do?
![image](https://user-images.githubusercontent.com/4089751/193809294-9eceb4d1-57ab-42c8-86d0-652b1d897a5a.png)

shawntanzk commented:
hmmm seems like clashing UBEORN IDs - you accidentally used UBERON:8480009 twice
fortunately this looks simple to fix - you need to give skin of breast another ID. 
I would quit the merge -> open up the edit file in protege -> change ID of skin of breast to something that isn't used -> try merging again. lmk if that works :)

paulacalipho commented:
@shawntanzk  If I try to open it from GitHub I have an error that repeats again and again. I can not open either from Protege. 

![image](https://user-images.githubusercontent.com/4089751/193814437-d9e54ab1-fccc-41a7-92e9-504de1ed1613.png)
![image](https://user-images.githubusercontent.com/4089751/193814655-78dcea9e-0a25-4aaf-80ae-50e213a77534.png)

shawntanzk commented:
oof probably best if i hop on a call with you and fix it together :) drop me a msg on slack and let me know when is good for you
